service.py

```diff
@@ -114,6 +114,42 @@ def createListItemFromFlatPlaylistItem(video):
 
     return listItem
 
+# get the index of the first video to be played in the submitted playlist url
+def playlistIndex(url, playlist):
+    if sys.version_info >= (3, 0):
+        from urllib.parse import urlparse, parse_qs
+    else:
+        from urlparse import urlparse, parse_qs 
+    
+    query = urlparse(url).query
+    queryParams = parse_qs(query)
+    
+    if 'v' not in queryParams:
+        print('no v param')
+        return None
+    
+    v = queryParams['v'][0]
+    
+    try:
```

Comment:
Why is this encapsulated in a try/except? As we can not assume there is always an `index` key?

Reply:
There might not be an index key, but it was introduced because the value might not be an integer. Then I thought that we don't really care which one it is...